### PR TITLE
Add weight to assessment questions

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/models/course/assessment.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@ class Course::Assessment < ActiveRecord::Base
   #   maximum grade.
   #   @return [Fixnum]
   calculated :maximum_grade, (lambda do
-    Course::Assessment::Question.select { sum(course_assessment_questions.maximum_grade) }.
+    Course::Assessment::Question.unscoped.select { sum(course_assessment_questions.maximum_grade) }.
```
Comment (lowjoel) [inline app/models/course/assessment.rb]: Why unscoped?

Comment (allenwq) [inline app/models/course/assessment.rb]: The default scope makes the calculated query unable to perform, this is the downside of default scope

Comment (allenwq) [inline app/models/course/assessment.rb]: I attached the errors here;

```
ActiveRecord::StatementInvalid: PG::GroupingError: ERROR:  column "course_assessment_questions.weight" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: ...essment_id" = "course_assessments"."id"  ORDER BY "course_as...
                                                             ^
: SELECT  "course_assessments".*, (SELECT sum("course_assessment_questions"."maximum_grade") FROM "course_assessment_questions" WHERE "course_assessment_questions"."assessment_id" = "course_assessments"."id"  ORDER BY "course_assessment_questions"."weight" ASC) AS maximum_grade FROM "course_assessments" WHERE "course_assessments"."id" = $1 LIMIT 1
```

Comment (lowjoel) [inline app/models/course/assessment.rb]: Then maybe you should change the question default scope to use a named scope, then try unscoping the named scope.

Comment (lowjoel) [inline app/models/course/assessment.rb]: ...or unscoping the where condition.

File: app/models/course/assessment.rb [hunk 2]
```diff
@@ -33,7 +33,8 @@ class Course::Assessment < ActiveRecord::Base
   #   maximum grade.
   #   @return [Fixnum]
   calculated :maximum_grade, (lambda do
-    Course::Assessment::Question.select { sum(course_assessment_questions.maximum_grade) }.
+    Course::Assessment::Question.unscope(:order).
```
Comment (allenwq) [inline app/models/course/assessment.rb]: @lowjoel I only unscope the `order` now. 

Happy new year !

**Comment by lowjoel** [comment]
:+1:

**Pull request merged**